cylc/flow/task_pool.py

```diff
@@ -2327,7 +2327,7 @@ def filter_task_proxies(
         ids: Iterable[str],
         warn_no_active: bool = True,
         inactive: bool = False,
-    ) -> 'Tuple[List[TaskProxy], Set[Tuple[str, PointBase]], List[str]]':
+    ) -> 'Tuple[List[TaskProxy], Set[Tuple[TaskDef, PointBase]], List[str]]':
```

Comment:
One thing to note with this change is that the hashability/equality of `TaskDef`s is purely reference based (i.e. `__hash__()` and `__eq__()` fall back to `object` default). This should be fine as the `TaskDef`s are always coming from the `scheduler.config.taskdefs` dictionary, I believe